src/plugins/z-bar.ts

```diff
@@ -0,0 +1,67 @@
+import { Plugin, Cordova } from './plugin';
+
+/**
+ * @name ZBar
+ * @description
+ * The ZBar Scanner Plugin allows you to scan 2d barcodes.
+ *
+ * Requires Cordova plugin: `cordova-plugin-cszbar`. For more info, please see the [zBar plugin docs](https://github.com/tjwoon/csZBar).
+ *
+ * @usage
+ * ```
+ * import { ZBar } from 'ionic-native';
+ *
+ * let zBarOptions = {
+ *       flash: "off",
+ *       drawSight: false
+ *     };
+ *
+ * ZBar.scan(zBarOptions, result => {
```

Comment:
After fixing the method below (by returning a promise), edit the usage to something like:
```
ZBar.scan(zBarOptions).then(result => { ... }).catch(error => { ... }); 
```